src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -541,10 +541,19 @@ public unsafe AssemblyInfo(string url, byte[] assembly, byte[] pdb)
             asmStream = new MemoryStream(assembly);
             peReader = new PEReader(asmStream);
             asmMetadataReader = PEReaderExtensions.GetMetadataReader(peReader);
+            Name = asmMetadataReader.GetAssemblyDefinition().GetAssemblyName().Name + ".dll";
+            AssemblyNameUnqualified = asmMetadataReader.GetAssemblyDefinition().GetAssemblyName().Name + ".dll";
```

Comment:
Looking at the implementation, `GetAssemblyDefinition()` returns a new object on every call. And `GetAssemblyName` constructing a new `AssemblyName` instance. so, we should avoid doing that twice.